src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9276,6 +9287,114 @@ def quantiles_along_axis0(
+        This greatly increases latency.
+
+        When the input frame has a single column, we can eliminate UNION ALL operations
+        by producing a single row where the columns are the different quantiles. Since there is
+        only a single row, we can do a relatively cheap UNPIVOT to make the result a single column.
+
+        >>> pd.Series([0, 1], name="b").quantile([0.25, 0.75])
+        +------+------+                                              +-------+------+
+        |   q1 |   q2 |                                              | index |    b |
+        +------+------+  -- UNPIVOT(b FOR quantile IN (q1, q2)) -->  +-------+------+
+        | 1.25 | 1.75 |                                              |    q1 | 1.25 |
+        +------+------+                                              +-------+------+
+                                                                     |    q2 | 1.75 |
+                                                                     +-------+------+
+
+        qcut() can drop the index column afterwards, but quantile() keeps it.
+        """
+        if index is not None:
+            # Snowpark UNPIVOT requires these to be strings
+            index = list(map(str, index))
+        original_frame = self._modin_frame
+        col_label = original_frame.data_column_pandas_labels[0]
+        col_ident = original_frame.data_column_snowflake_quoted_identifiers[0]
+        new_labels = [str(quantile) for quantile in q]
+        new_idents = (
+            original_frame.ordered_dataframe.generate_snowflake_quoted_identifiers(
+                pandas_labels=new_labels
+            )
+        )
+        ordered_dataframe = original_frame.ordered_dataframe.agg(
+            *[
+                column_quantile(col(col_ident), interpolation, quantile).as_(new_ident)
+                for new_ident, quantile in zip(new_idents, q)
+            ]
+        )
+        index_ident = ordered_dataframe.generate_snowflake_quoted_identifiers(
+            pandas_labels=[None]
+        )[0]
+        # In order to preserve index labels, we call unpivot directly instead of using transpose_single_row
+        ordered_dataframe = ordered_dataframe.unpivot(
+            original_frame.data_column_snowflake_quoted_identifiers[0],
+            index_ident,
+            new_idents,
+            col_mapper=dict(zip(new_idents, index))
+            if index is not None
+            else dict(zip(new_idents, new_labels)),
+        ).ensure_row_position_column()
+        internal_frame = InternalFrame.create(
+            ordered_dataframe=ordered_dataframe,
+            data_column_pandas_labels=[col_label],
+            data_column_pandas_index_names=[None],
+            data_column_snowflake_quoted_identifiers=[col_ident],
+            index_column_pandas_labels=[None],
+            index_column_snowflake_quoted_identifiers=[index_ident],
+        )
+        if index_dtype:
```

Comment:
you seem to call this function never with `index_dtype=None` but always with a valid index_dtype (the None case seems to become `index_dtype=index_dtype or float` above in `quantiles_along_axis0`. Is there a scenario where `index_dtype=None` is called? If not, can we remove the if statement, and adjust the index_dtype type hint?

Reply:
I've changed index_dtype to default to `float` now, and removed this `if index_dtype` guard.